internal/host/validator.go

```diff
@@ -1432,3 +1432,40 @@ func (v *validator) printNonOverlappingSubnets(c *validationContext, status Vali
 	}
 	return fmt.Sprintf("Unexpected status %s", status)
 }
+
+func (v *validator) isVSphereDiskUIDEnabled(c *validationContext) ValidationStatus {
+	if c.inventory == nil {
+		return ValidationPending
+	}
+
+	if c.cluster == nil {
+		return ValidationPending
+	}
+
+	if c.cluster.Platform == nil || c.cluster.Platform.Type == nil || *c.cluster.Platform.Type != models.PlatformTypeVsphere {
+		return ValidationSuccess
+	}
+
+	if c.inventory.Disks == nil {
+		return ValidationPending
+	}
+
+	for _, disk := range c.inventory.Disks {
+		if v.hwValidator.IsValidStorageDeviceType(disk) && !disk.HasUUID {
```

Comment:
```suggestion
                // vSphere only adds a UUID to disks which can potentially be used for storage,
                // if any of them doesn't have that flag, it's likely because the user has forgot to
                // enable `disk.EnableUUID` for this virtual machine
                // See https://access.redhat.com/solutions/4606201
		if v.hwValidator.IsValidStorageDeviceType(disk) && !disk.HasUUID {
```
This is not so obvious